tests/test_privacy_scan_review.py

```
@@ -0,0 +1,407 @@
+
+            def open_then_replace(path: os.PathLike[str] | str, flags: int) -> int:
+                descriptor = real_open(path, flags)
+                if Path(path).resolve() == source.resolve():
+                    os.replace(replacement, source)
+                return descriptor
+
+            with patch.object(_review.os, "open", side_effect=open_then_replace):
+                with self.assertRaisesRegex(
+                    ReviewRecordError,
+                    "owner-review finding file mode changed",
+                ):
```

Ruff (0.16.2)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)